### PR TITLE
chore: add eslint rule to check just the re-exported functions have jsdocs

### PR DESCRIPTION
**Pull request by moldy530**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR adds a new ESLint rule, updates TypeScript configuration, refactors class comments, and enhances logger documentation.

### Detailed summary
- Added `eslint-plugin-eslint-rules` to `package.json`
- Updated TypeScript configuration in `tsconfig.json`
- Refactored class comments in various files
- Enhanced documentation for the `Logger` class

> The following files were skipped due to too many changes: `site/pages/migration-guides/migrating-to-v3.md`, `aa-sdk/core/src/errors/base.ts`, `account-kit/signer/src/client/base.ts`, `account-kit/react/src/hooks/useUser.ts`, `account-kit/core/src/actions/getSignerStatus.ts`, `.github/workflows/on-pull-request.yml`, `aa-sdk/core/src/middleware/defaults/paymasterAndData.ts`, `aa-sdk/core/src/client/decorators/bundlerClient.ts`, `aa-sdk/core/src/client/decorators/smartAccountClient.ts`, `account-kit/infra/src/defaults.ts`, `aa-sdk/core/src/errors/entrypoint.ts`, `aa-sdk/core/src/errors/transaction.ts`, `account-kit/core/src/actions/watchSigner.ts`, `account-kit/react/src/hooks/useLogout.ts`, `account-kit/core/src/actions/watchChain.ts`, `account-kit/infra/src/client/decorators/smartAccount.ts`, `account-kit/core/src/actions/watchConnection.ts`, `account-kit/infra/src/client/smartAccountClient.ts`, `.eslintrc`, `account-kit/infra/src/client/isAlchemySmartAccountClient.ts`, `account-kit/smart-contracts/src/msca/plugins/multisig/index.ts`, `account-kit/infra/src/chains.ts`, `account-kit/react/src/hooks/useAddPasskey.ts`, `account-kit/react/src/hooks/useSignMessage.ts`, `account-kit/smart-contracts/src/msca/plugins/session-key/extension.ts`, `aa-sdk/ethers/src/account-signer.ts`, `account-kit/react/src/hooks/useExportAccount.ts`, `account-kit/core/src/actions/watchSignerStatus.ts`, `account-kit/react/src/hooks/useSignTypedData.ts`, `account-kit/core/src/actions/watchUser.ts`, `account-kit/core/src/actions/getAccount.ts`, `account-kit/smart-contracts/src/msca/plugins/multisig/extension.ts`, `account-kit/core/src/actions/watchBundlerClient.ts`, `account-kit/react/src/hooks/useAuthenticate.ts`, `account-kit/smart-contracts/src/msca/account-loupe/decorator.ts`, `account-kit/smart-contracts/src/msca/plugins/multi-owner/extension.ts`, `templates/eslint/package.json`, `account-kit/core/src/actions/watchAccount.ts`, `account-kit/react/src/hooks/useAccount.ts`, `aa-sdk/core/src/middleware/defaults/userOpSigner.ts`, `account-kit/react/src/hooks/useSmartAccountClient.ts`, `account-kit/core/src/actions/createAccount.ts`, `account-kit/infra/src/middleware/userOperationSimulator.ts`, `account-kit/smart-contracts/src/light-account/decorators/lightAccount.ts`, `account-kit/smart-contracts/src/msca/client/alchemyClient.ts`, `account-kit/smart-contracts/src/light-account/clients/alchemyClient.ts`, `account-kit/infra/src/middleware/feeEstimator.ts`, `account-kit/smart-contracts/src/index.ts`, `account-kit/react/src/hooks/useDropAndReplaceUserOperation.ts`, `account-kit/smart-contracts/src/msca/errors.ts`, `account-kit/smart-contracts/src/light-account/clients/multiOwnerAlchemyClient.ts`, `account-kit/smart-contracts/src/light-account/decorators/multiOwnerLightAccount.ts`, `aa-sdk/core/src/errors/client.ts`, `account-kit/infra/src/actions/simulateUserOperationChanges.ts`, `account-kit/react/src/hooks/useWaitForUserOperationTransaction.ts`, `account-kit/smart-contracts/src/light-account/accounts/multiOwner.ts`, `account-kit/smart-contracts/src/light-account/accounts/account.ts`, `account-kit/infra/src/gas-manager.ts`, `account-kit/smart-contracts/src/msca/plugin-manager/installPlugin.ts`, `account-kit/smart-contracts/src/light-account/actions/updateOwners.ts`, `aa-sdk/core/src/actions/smartAccount/buildUserOperation.ts`, `aa-sdk/core/src/actions/smartAccount/sendTransactions.ts`, `aa-sdk/core/src/actions/smartAccount/dropAndReplaceUserOperation.ts`, `account-kit/infra/src/client/rpcClient.ts`, `aa-sdk/core/src/actions/smartAccount/sendTransaction.ts`, `account-kit/smart-contracts/src/msca/client/multiSigAlchemyClient.ts`, `account-kit/smart-contracts/src/msca/plugin-manager/decorator.ts`, `account-kit/smart-contracts/src/msca/plugins/multisig/utils/getSignerType.ts`, `account-kit/smart-contracts/src/light-account/clients/multiOwnerLightAccount.ts`, `account-kit/smart-contracts/src/light-account/clients/client.ts`, `aa-sdk/core/src/actions/smartAccount/waitForUserOperationTransacation.ts`, `account-kit/smart-contracts/src/msca/plugins/multisig/utils/combineSignatures.ts`, `aa-sdk/core/src/middleware/defaults/feeEstimator.ts`, `aa-sdk/core/src/client/bundlerClient.ts`, `account-kit/smart-contracts/src/light-account/actions/transferOwnership.ts`, `account-kit/smart-contracts/src/msca/account/multiOwnerAccount.ts`, `account-kit/core/src/store/client.ts`, `aa-sdk/core/src/client/smartAccountClient.ts`, `templates/eslint/src/resolveReExport.ts`, `account-kit/infra/src/client/internal/smartAccountClientFromRpc.ts`, `templates/eslint/src/fixers/re-export-fixer.ts`, `account-kit/smart-contracts/src/msca/account/multisigAccount.ts`, `account-kit/smart-contracts/src/msca/plugins/session-key/utils.ts`, `aa-sdk/core/src/ens/utils.ts`, `templates/eslint/src/rules/jsdoc-reexported-functions.ts`, `account-kit/smart-contracts/src/msca/client/client.ts`, `aa-sdk/core/src/utils/index.ts`, `account-kit/infra/src/client/decorators/alchemyEnhancedApis.ts`, `aa-sdk/core/src/errors/account.ts`, `aa-sdk/core/src/entrypoint/index.ts`, `account-kit/smart-contracts/src/msca/utils.ts`, `aa-sdk/core/src/signer/utils.ts`, `templates/eslint/src/fixers/openai-worker.ts`, `aa-sdk/core/src/account/smartContractAccount.ts`, `yarn.lock`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: templates/eslint/src/rules/jsdoc-reexported-functions.ts
Comment (rthomare) [reply templates/eslint/src/rules/jsdoc-reexported-functions.ts:22]: ironic

**Review by rthomare — APPROVED** [review]

File: templates/eslint/src/rules/jsdoc-reexported-functions.ts
Comment (moldy530) [reply templates/eslint/src/rules/jsdoc-reexported-functions.ts:22]: LOL yea I should comment these methods in actuality because this needs docs for maintainability

**Review by rthomare — APPROVED** [review]

**Review by dphilipson — APPROVED** [review]
I really appreciate your doing things like this. Most developers would just accept whatever behavior already exists in eslint, you're really above and beyond for doing things like this.

**Comment by moldy530** [comment]
heads up, gonna hold off on merging this until I've got https://github.com/alchemyplatform/aa-sdk/pull/728 working and then worked my way through batches of backfills of jsdoc comments.

once this branch passes linting, I'll merge it into v4.x.x. Then I'll get the CLI merged in

**Pull request merged**
